orangecanvas/canvas/scene.py

```diff
@@ -493,10 +493,6 @@ def channel_name(channel):
         sink_name = channel_name(sink_channel)
 
         fmt = "<b>{0}</b>&nbsp; \u2192 &nbsp;<b>{1}</b>"
-        item.setToolTip(
-            fmt.format(escape(source_name),
-                       escape(sink_name))
-        )
```

Comment:
Don't remove the tool tip.

The channel name on the link can be obscured, in a crowded spot, not even in view, just hard to read (due to rotation) and not where the mouse pointer is (and hence the users eyes).

Reply:
@markotoplak, @ajdapretnar, thoughts?

Reply:
Yes, it is a good point. We have to consider complex workflows, too. Such as, what happens in this case:
<img width="183" alt="Screen Shot 2020-09-25 at 11 28 58" src="https://user-images.githubusercontent.com/12524972/94251064-55b33400-ff22-11ea-8f40-542129bbd6ed.png">
I agree we should keep the tooltip.

Reply:
I like this new feature a lot. 

However, I would say that having both the tooltip (which appears a bit too slow I think, but that is another discussion) and the channel names is redundant. 

For @ajdapretnar's example, if they wouldn't fit anyway, the channel names should disappear automatically and the tooltips should remain, although even then it could be hard to identify what belong where.

So, is it possible to display either the tooltip or the hover-channel-names depending on how crowded the widgets are?

Reply:
I agree with Borondics, it's redundant. TBH, I'd like to get rid of all tooltips on canvas, and convey information via hover.

@ajdapretnar, do such connections really occur in complex workflows? I assume overlaying widgets don't, and shouldn't the workflow extend horizontally in such cases?

What I can do is make the currently hovered item have the highest Z-value, and give it a white background. Bad readability due to rotation could be remedied by putting the channel name on the other side of the link, should it otherwise appear upside down.

Reply:
Setting z-value to something big could be enough. These cases should be rare and it is still possible to double-click a connection (probably faster than waiting for the tooltip).

Reply:
@irgolic it does come up in complex workflows and sometimes it is not practical to extend horizontally. This was a reason that we added continuous zoom to the canvas. Have a look at what some of our users build at https://github.com/biolab/orange3/issues/4910.

Reply:
@borondics Does this work for you in a more complex workflow?